### PR TITLE
Allow configuring MaxWorkflowRetentionInDays

### PR DESCRIPTION
**Pull request by longquanzheng**
**What changed?**
Allow configuring MaxWorkflowRetentionInDays

**Why?**
For https://github.com/uber/cadence/issues/3981

**How did you test it?**
Local test the dynamic config is loaded. 

**Potential risks**
No

### CODE REVIEW

**Comment by longquanzheng** [comment]
Github suggests you to review the code :P Can you help take a look? @mantas-sidlauskas @Groxx Thanks

File: service/worker/scanner/history/scavenger.go [hunk 1]
```diff
@@ -96,18 +100,20 @@ func NewScavenger(
 	hbd ScavengerHeartbeatDetails,
 	metricsClient metrics.Client,
 	logger log.Logger,
+	maxWorkflowRetentionInDays int,
```
Comment (yycptt) [inline service/worker/scanner/history/scavenger.go:103]: nit: shall we make this really dynamic and doesn't require worker restart upon changing the value?

Comment (longquanzheng) [reply service/worker/scanner/history/scavenger.go:103]: good point.

Comment (longquanzheng) [reply service/worker/scanner/history/scavenger.go]: Done. Thanks for the comments :D

File: service/worker/scanner/history/scavenger_test.go [hunk 2]
```diff
@@ -67,7 +69,10 @@ func (s *ScavengerTestSuite) createTestScavenger(rps int) (*mocks.HistoryV2Manag
 	db := &mocks.HistoryV2Manager{}
 	controller := gomock.NewController(s.T())
 	workflowClient := history.NewMockClient(controller)
-	scvgr := NewScavenger(db, 100, workflowClient, ScavengerHeartbeatDetails{}, s.metric, s.logger)
+
+	dynamicConfigColleciton := dynamicconfig.NewNopCollection()
+	maxWorkflowRetentionInDays := dynamicConfigColleciton.GetIntProperty(0, domain.DefaultMaxWorkflowRetentionInDays)
```
Comment (yycptt) [inline service/worker/scanner/history/scavenger_test.go:74]: nit: use `dc.GetIntPropertyFn(common.DefaultMaxWorkflowRetentionInDays)` in test so no collection is needed.

**Review by yycptt — APPROVED** [review]

File: service/worker/scanner/history/scavenger_test.go (hunk 2 above)
Comment (longquanzheng) [reply service/worker/scanner/history/scavenger_test.go]: Sure, thanks!

**Pull request merged**